### PR TITLE
Problem #18: Add button to close the editor.

### PR DESCRIPTION
**Pull request by kegor**
**Problem https://github.com/Starcounter/Blending/issues/18:**
>Today we made a decision to:
> - add a "Close" button at the right the "Save" button
> - clicking on "Close" should cancel pending changes and close the composition editor panel
> - remove the "Cancel" button

**Solution:**
Close button was added, Cancel button was removed.

_Note:_
For the moment I copied two functions from https://github.com/Starcounter/Blending/blob/master/src/BlendingEditor/wwwroot/BlendingEditor/elements/compositioneditor-inserter/compositioneditor-inserter.html because I didn't find a method to reuse them. Please ping me if you have an idea.

**Screenshots:**
close_button

### CODE REVIEW

**Comment by tomalec** [comment]
What was the reason to drop revert feature?

File: starcounter-layout-html-editor.html [hunk 1]
```diff
@@ -349,17 +349,37 @@
...
 
+                        return this.partial.shadowRoot;
+                    }
+                    
+                    closePanel();
                 }
             });
 
+            function closePanel() {
+                var panel = document.querySelector("compositioneditor-panel");
```
Comment (tomalec) [inline starcounter-layout-html-editor.html]: This makes a hidden dependency on `compositioneditor-panel` and `compositioneditor-inserter`.

**Comment by kegor** [comment]
Hi @tomalec.
@warpech @PatrykSzwer and I decided that we would like to add `Close` button mostly for more friendly UX. And because 4 buttons look too much we decided to remove `Cancel` button.
I hope guys will correct me if I'm wrong somewhere.

File: starcounter-layout-html-editor.html (hunk 1 above)
Comment (kegor) [reply starcounter-layout-html-editor.html]: Hi @tomalec, I've made related changes here and in newly created [PR in Blenindg](https://github.com/Starcounter/Blending/pull/167), how they look for you?

**Comment by tomalec** [comment]
@kegor  could you release a new version of the element, with those changes?

File: starcounter-layout-html-editor.html [hunk 2]
```diff
@@ -349,14 +349,19 @@
...
-                    this.partial._compositionChanged(this.partial.storedLayout)
-                    return this.partial.shadowRoot;
+                close: function() {
+                    var closeEvent = new Event('close');
 
+                    if (this.partial) {
+                        this.partial._forceLayoutChange = true;
+                        this.partial._compositionChanged(this.partial.storedLayout)
+
+                        window.dispatchEvent(closeEvent);
```
Comment (tomalec) [inline starcounter-layout-html-editor.html]: You should dispatch the event on `this` not on `window`

Comment (kegor) [reply starcounter-layout-html-editor.html]: Hi @tomalec quick fix was pushed. Сould you please expand my horizons and clarify why here `this` should be used, I saw dispatching on `window` at this line: https://github.com/Starcounter/starcounter-layout-html-editor/blob/69d587d5d0ef0d1342c37e6a49c9e50e0a8663fe/starcounter-layout-html-editor.html#L303 here this is valid?

Comment (tomalec) [reply starcounter-layout-html-editor.html]: Sure :) `resize` is is mostly a hack for ACE editor, that listens to resizing the window only.

**Pull request merged**